File: 2RingEmployeesMgmt/Domain.Services/Services/PositionService.cs
using Domain.Models;
using Domain.Services.Interfaces;
using Domain.Services.Validation;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain.Services
{
    public class PositionService : IPositionService
    {
        private readonly IUnitOfWork _dbContext;

        public PositionService(IUnitOfWork dbContext)
        {
            _dbContext = dbContext;
        }

        public Task AddNewPosition(Position position)
        {
            _dbContext.Add(position);
            return _dbContext.SaveChangesAsync();
        }

        public async Task<bool> DeletePosition(Position position)
        {
            _dbContext.Delete(position);
            var affectedRows = await _dbContext.SaveChangesAsync();
            return affectedRows != 0;
        }

        public async Task<IReadOnlyCollection<Position>> GetAllPositions()
        {
            var positions = await _dbContext.Query<Position>().ToListAsync();
            return positions;
        }

        public async Task<Position> GetPosition(int id)
        {
            var position = await _dbContext.Query<Position>().Where(p => p.Id == id).FirstOrDefaultAsync();
            return position;
        }
    }
}

File: 2RingEmployeesMgmtFE/2ring-employees-mgmt/src/services/PositionService.ts
import axios from "axios";
import Position from "@/models/Position";

class PositionService {
  async getAllPositions(): Promise<Position[]> {
    const res = await axios.get<Position[]>(`EmployeePosition`);
    return res.data.map((pos) => new Position(pos));
  }
  async addPosition(position: Position): Promise<Position> {
    const res = await axios.post<Position>(`EmployeePosition`, position);
    return new Position(res.data);
  }
  deletePosition(id: Number): Promise<void> {
    return axios.delete(`EmployeePosition/${id}`);
  }
}

export default new PositionService();
